packages/client/src/types/signature.ts

```diff
@@ -32,18 +33,45 @@ export type SignatureMaterial = {
 
 export type SignerFunc = (payload: Buffer) => Promise<SignatureMaterial>;
 
-export function extractSignatureMaterial(
-  dsseEnvelope: Envelope,
-  publicKey: string
-): SignatureMaterial {
-  const signature = dsseEnvelope.signatures[0];
-
-  return {
-    signature: signature.sig,
-    key: {
-      id: signature.keyid,
-      value: publicKey,
-    },
-    certificates: undefined,
-  };
+type CallbackSignerOptions = {
+  signer: SignerFunc;
+};
+
+// Adapter to allow the legacy SignerFunc callback to be used as a new Signer
+// interface.
+export class CallbackSigner implements Signer {
```

Comment:
Should this class live in the types directory? Wondering if it will get lost here